src/Writers/LaTeXWriter.jl

```diff
@@ -83,23 +91,65 @@ function render(doc::Documents.Document)
                 writefooter(context, doc)
             end
             cp(STYLE, "documenter.sty")
-            if hastex()
-                outdir = joinpath(doc.user.root, doc.user.build)
-                pdf = replace("$(doc.user.sitename).pdf", " " => "")
-                try
-                    run(`latexmk -f -interaction=nonstopmode -view=none -lualatex -shell-escape $file`)
-                catch err
-                    @warn "failed to compile. Check generated LaTeX file."
-                    cp(file, joinpath(outdir, file); force = true)
-                end
-                cp(pdf, joinpath(outdir, pdf); force = true)
-            else
-                @warn "`latexmk` and `lualatex` required for PDF generation."
-            end
+
+            # compile .tex and copy over the .pdf file if compile_tex return true
+            status = compile_tex(doc, fmt, texfile)
+            status && cp(pdffile, joinpath(doc.user.root, doc.user.build, pdffile); force = true)
+        end
+    end
+end
+
+const DOCKER_IMAGE_TAG = "0.1"
+
+function compile_tex(doc, fmt, texfile)
+    # engine = get(ENV, "DOCUMENTER_LATEX_ENGINE", "docker") # TODO: make this configurable from makedocs
+    engine = get(fmt.info, "engine", "docker")
+    if engine == "latexmk"
+        Sys.which("latexmk") === nothing && (@error "LaTeXWriter: latexmk command not found."; return false)
+        @info "LaTeXWriter: using latexmk to compile tex."
+        try
+            piperun(`latexmk -f -interaction=nonstopmode -view=none -lualatex -shell-escape $texfile`)
+            return true
```

Comment:
With latexmk, the PDF is not getting copied back to `build/` at the moment, right?

Reply:
Why not? https://github.com/JuliaDocs/Documenter.jl/pull/891/files#diff-862e2c1cb8b709bb81be9bc357122322R97

(only if the compile was succesful though)

Reply:
My bad, I didn't read the code carefully enough. And indeed, I had a non-successful compile.